[OU-FIX] product: avoid reserved SQL keyword "sequence"

MaPaLo76 opened the pull request:
When running OpenUpgrade 18.0 on PostgreSQL, queries using  `row_number(...) sequence` raise a syntax error due to unquoted alias names with postgres version 13.0

This PR fixes:
- `sequence` → `"sequence"`

to ensure compatibility with PostgreSQL reserved keywords and avoid `syntax error at or near ...`.

This patch unblocks migration from Odoo 17.0 to 18.0 for affected modules (, `product`) with postgres 13.0.

MaPaLo76 committed:
Update post-migration.py

When running OpenUpgrade 18.0 on PostgreSQL, queries using  `row_number(...) sequence` raise a syntax error due to unquoted alias names.

This PR fixes:
- `sequence` → `"sequence"`

to ensure compatibility with PostgreSQL reserved keywords and avoid `syntax error at or near ...`.

This patch unblocks migration from Odoo 17.0 to 18.0 for affected modules (, `product`).

pedrobaeza commented:
Maybe it's better to simply not call it `sequence`, but `sub` or any other name.

hbrunn reviewed (changes requested):
please write a commit message describing what the change does (it has nothing to do with json_object_agg btw) and remove the trailing semicolon.

the patch also contains a lot of unnecessary changes, but i don't mind them.

you might also consider @pedrobaeza's proposal to just not use a non-reserved keyword here

MiquelRForgeFlow commented:
I also think the best way is not using the reserved keywords.

Also, commit title should be `[OU-FIX] product: brief explanation`.

MaPaLo76 commented:
sorry for the inconvenience, i made a mistake during this PR. I thought I was in my fork.

I'm relatively inexperienced at submitting pull requests. This is my second one. The first one, in openupgradelib, worked better.
Do I need to do anything else?

MaPaLo76 committed:
[OU-FIX] product: avoid reserved SQL keyword "sequence"

Renamed the alias "sequence" to "seq" in the post-migration script
to prevent SQL syntax issues with reserved keywords.

MiquelRForgeFlow commented on openupgrade_scripts/scripts/product/18.0.1.2/post-migration.py:
```diff
@@ -12,21 +12,21 @@ def product_document_sequence(env):
         """
         UPDATE product_document
-        SET sequence=sequence.sequence
+        SET sequence=seq.sequence
```
```suggestion
        SET sequence=seq.seq
```

MaPaLo76 committed:
[OU-FIX] Rename alias to avoid reserved keyword usage in SQL

Thanks for the suggestion – applied!

Co-authored-by: Miquel Raïch <miquel.raich@forgeflow.com>
